### PR TITLE
Feature/updated-checker-app-apis

### PR DESCRIPTION
**Pull request by sarge1989**
Implemented new API routes and interfaces to accomodate the various screens

With these interfaces:

GET /checkers/:checkerId should be called upon login/dashboard page
GET /checkers/:checkerId/votes should be called upon my votes page
GET /messages/:messageId/voteRequests/:voteRequestId should be called when entering the individual vote page
PATCH /messages/:messageId/voteRequests/:voteRequestId when submitting votes

See https://1drv.ms/p/s!An94l4x7iqxgocQwWzZzFZzuv2qrnA?e=5UcieB for screenshots of the various pages for reference

### CODE REVIEW

**Comment by sarge1989** [comment]
@btjl 

Some comments on api.ts

For
- GET /checkerData/:id > able to use my GET /checkers/:checkerId instead
- PUT /checkerData/:id > able to rename to PUT /checkers/:checkerId to follow the general pattern of how I name API routes? Actually for consistency, can we use POST instead?

File: functions/src/definitions/api/interfaces.ts [hunk 1]
```diff
@@ -25,4 +25,64 @@ interface createChecker {
...
+interface Checker {
+  name: string
+  type: "human" | "ai"
+  isActive: boolean | null
+  pendingVoteCount: number
+  last30days: last30DaysStats
+  achievements: Achievements | null
+  level: number
+  experience: number
+}
```
Comment (btjl) [inline functions/src/definitions/api/interfaces.ts:37]: Referencing `functions/src/types.ts` there seem to be missing properties from this interface

```typescript
export type Checker = {
  name: string
  type: "human" | "ai"
  isActive: boolean | null
  isOnboardingComplete: boolean | null
  singpassOpenId: string | null
  telegramId: string | null
  whatsappId?: string | null
  level: number
  experience: number
  numVoted: number
  voteWeight: number
  numCorrectVotes: number
  numVerifiedLinks: number
  preferredPlatform: string | null
  lastVotedTimestamp: Timestamp | null
  getNameMessageId: string | null
}
```

File: functions/src/definitions/api/interfaces.ts [hunk 2]
```diff
@@ -25,4 +25,64 @@ interface createChecker {
...
+  totalVoted: number
+  accuracyRate: number
+  averageResponseTime: number
+  peopleHelped: number
+}
+
+interface Achievements {
+  //TODO: add more achievements in future
+}
+
```
Comment (btjl) [inline functions/src/definitions/api/interfaces.ts:73]: Is this a valid interface as well?

```suggestion
export type VoteRequest = {
  factCheckerDocRef: DocumentReference
  platformId: string | null
  platform: "whatsapp" | "telegram" | "agent"
  hasAgreed: boolean | null
  triggerL2Vote: boolean | null
  triggerL2Others: boolean | null
  sentMessageId: string | null
  vote: 0 | 1 | 2 | 3 | 4 | 5 | null
  category: string | null
  createdTimestamp: Timestamp | null
  acceptedTimestamp: Timestamp | null
  votedTimestamp: Timestamp | null
}
```

File: functions/src/definitions/api/interfaces.ts (hunk 1 above)
Comment (sarge1989) [reply functions/src/definitions/api/interfaces.ts:37]: Am intentionally returning less data from the API interface than is stored in the DB. Cos it's only for the purposes of the webapp, so some of these are unnecessary.

File: functions/src/definitions/api/interfaces.ts (hunk 2 above)
Comment (sarge1989) [reply functions/src/definitions/api/interfaces.ts:73]: For the purposes of the webapp, the API will expose a "vote" interface that is a combination of voteRequest and message. Basically for this interface I'm coming from the perspective of the checkers app, not so much the database schema.

File: .vscode/settings.json
Comment (sarge1989) [inline .vscode/settings.json:1]: Can remove this from git? It changed my settings.

Comment (btjl) [reply .vscode/settings.json:1]: Yeap, should git ignore these

**Pull request merged**